src/dotnet-scaffolding/dotnet-scaffold-aspnet/Helpers/BlazorCrudHelper.cs

```diff
@@ -193,17 +212,64 @@ private static async Task<BlazorCrudAppProperties> GetBlazorPropertiesAsync(Docu
             }
         }
 
-        if (appRazorDocument != null)
+        // Check for Routes.razor first (prioritize it), then App.razor
+        blazorAppProperties.HasRoutesRazor = routesRazorDocument != null;
+        blazorAppProperties.HasAppRazor = appRazorDocument != null;
+        
+        var routerDocument = routesRazorDocument ?? appRazorDocument;
```

Comment:
`Routes.razor` is not the same as `App.razor`. `Routes.razor` is typically defined inside `App.razor` and only contains mainly the `<Router>` definition. You should not apply any `App.razor` logic to `Routes.razor`